src/main/java/Duke.java

```diff
@@ -1,10 +1,244 @@
+                        }
+                    } else if (test[0].equals("delete")) {
+                        try {
+                            if (lists.size() == 0) {
+                                throw new IndexOutOfBoundsException();
+                            }
+                            if (test.length == 1) {
+                                throw new MissingNumberException();
+                            }
+                            int idx = Integer.parseInt(test[1]) - 1;
+                            Response msg = new Response(new Task[]{lists.get(idx)}, Response.Tag.REMOVE, lists.size() - 1);
+                            System.out.println(msg.getResponse());
+                            lists.remove(idx);
+                        } catch (IndexOutOfBoundsException e) {
+                            Response msg = new Response(new String[]{"The number does not exist!"});
+                            System.out.println(msg.getResponse());
+                        } catch (MissingNumberException e) {
+                            Response msg = new Response(new String[]{"Please select the task that you want to delete!"});
+                            System.out.println(msg.getResponse());
+                        }
+                    } else if (test[0].equals("find")) {
+                        try {
+                            if (test.length == 1) {
+                                throw new DukeException();
+                            }
+                            ArrayList<Task> satisfiedTasks = new ArrayList<>();
+                            for (int i = 0; i < lists.size(); i++) {
+                                if (lists.get(i).task.contains(test[1])) {
+                                    satisfiedTasks.add(lists.get(i));
+                                }
+                            }
+                            Response msg = new Response(satisfiedTasks.toArray(new Task[0]), Response.Tag.FIND);
+                            System.out.println(msg.getResponse());
+                        } catch (DukeException e) {
+                            Response msg = new Response(new String[]{"Please include the keyword!"});
+                            System.out.println(msg.getResponse());
+                        }
+                    } else {
+                        if (test[0].equals("todo")) {
+                            try {
+                                String description = received.split("todo ")[1];
+                                Todo todo = new Todo(description);
+                                lists.add(todo);
+                                Response msg = new Response(new Task[]{todo}, Response.Tag.ADD, lists.size());
+                                System.out.println(msg.getResponse());
+                            } catch (ArrayIndexOutOfBoundsException err) {
+                                MissingDescriptionException realError = new MissingDescriptionException("a todo");
```

Comment:
Perhaps it would be better if you throw all the Exception? They might be useful in the future use (for try-catching).  I noticed the same issue in several other places too.